src/org/python/core/PyMemoryView.java

```diff
@@ -67,7 +67,8 @@ static PyObject memoryview_new(PyNewWrapper new_, boolean init, PyType subtype,
         ArgParser ap = new ArgParser("memoryview", args, keywords, "object");
         PyObject obj = ap.getPyObject(0);
 
-        if (obj instanceof BufferProtocol) {
+        // memoryview in 2.7 throws typeerror on unicode rather than assuming encoding like buffer().
```

Comment:
Not this way. It's important that we accept any (byte) buffer object. Try ``n = memoryview(buffer(b'abc'))``. Solve by defining ``PyUnicode.getBuffer()`` to throw?